### PR TITLE
Lint: Fix str-format-in-logging, len-as-condition, etc. (6)

### PR DESCRIPTION
**Pull request by terminalmage**
This fixes violations found in the following locations:

- salt/state.py
- salt/states/
- salt/tops/
- salt/transport/

### CODE REVIEW

File: salt/states/dvs.py [hunk 1]
```diff
@@ -608,7 +605,7 @@ def portgroups_configured(name, dvs, portgroups):
                                 {'old': current_pg}})
         __salt__['vsphere.disconnect'](si)
     except salt.exceptions.CommandExecutionError as exc:
-        log.error('Error: {0}\n{1}'.format(exc, traceback.format_exc()))
+        log.traceback('Encountered error')
```
Comment (s0undt3ch) [inline salt/states/dvs.py]: `log.exception()`?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/states/dvs.py (hunk 1 above)
Comment (terminalmage) [reply salt/states/dvs.py]: d'oh, yes

Comment (terminalmage) [reply salt/states/dvs.py]: Should be fixed now.

**Review by s0undt3ch — APPROVED** [review]

**Comment by terminalmage** [comment]
I fixed some incorrect mocking that should resolve the mac test failures.

File: salt/states/btrfs.py [hunk 2]
```diff
@@ -119,8 +119,8 @@ def wrapper(*args, **kwargs):
             kwargs['__dest'] = dest
             ret = action(*args, **kwargs)
         except Exception as e:
-            log.error('''Traceback: {}'''.format(traceback.format_exc()))
-            ret['comment'].append(e)
+            log.exception('Encountered error mounting %s', device)
+            ret['comment'].append(six.text_type(e))
```
Comment (aplanas) [inline salt/states/btrfs.py]: You introduced two error here:

* This error is not a mount error, is the exception produced in the method that is wrapping this code (the call from line 120). The mount error was taken care a bit before.

* 'e' do not contain the traceback, and we do not want to hide it. We need to print it in the log at least (and maybe in the comments too)

I will change that in a different pull request, and I will integrate your fix for log.exception and six.text_type

Comment (terminalmage) [reply salt/states/btrfs.py:123]: >     * This error is not a mount error, is the exception produced in the method that is wrapping this code (the call from line 120). The mount error was taken care a bit before.

OK, so the message in the log was inaccurate, fair enough.

>     * 'e' do not contain the traceback, and we do not want to hide it. We need to print it in the log at least (and maybe in the comments too)

I have no idea what you're talking about. Nothing was hidden. Logging at `exception` level is the same thing as doing `log.error('some message', exc_info=True)`. It includes the exception.

**Comment by terminalmage** [comment]
Ported to master in https://github.com/saltstack/salt/pull/57443

**Pull request merged**